doc/pfc_asym/PFC_Asymmetric_HLD.md

```diff
@@ -0,0 +1,391 @@
+# Asymmetric PFC Test Plan
+
+* [Overview](#Overview)
+   * [Scope](#Scope)
+   * [Testbed](#Testbed)
+* [Setup configuration](#setup-configuration)
+* [Existed modules refactoring](#existed-modules-refactoring)
+* [Python  modules to setup and run test](#python-modules-to-setup-and-run-test)
+   * [Python modules](#python-modules)
+   * [Global variables](#global-variables)
+   * [PTF test case execution](#ptf-test-case-execution)
+   * [Pytest fixtures](#pytest-fixtures)
+		* [deploy_pfc_gen](#deploy_pfc_gen)
+		* [setup](#setup)
+		* [pfc_storm_template](#pfc_storm_template)
+		* [pfc_storm_runner](#pfc_storm_runner)
+		* [enable_pfc_asym](#enable_pfc_asym)
+* [Test](#Test)
+* [Test cases](#Test-cases)
+
+## Overview
+The purpose is to test functionality of Asymmetric PFC on the SONIC based DUT, closely resembling production environment.
+
+### Scope
+The test is targeting a running SONIC system with fully functioning configuration. The purpose of the test is to perform functional testing of Asymmetric PFC on SONIC system. There will be reused existed PTF test suite for PFC Asymmetric which is located at ```ansible\roles\test\files\saitests\pfc_asym.py```.
+
+### Testbed
+The test will run on the following testbeds:
+* T0
+
+## Setup configuration
+No setup pre-configuration is required, test will configure and clean-up all the configuration.
+
+## Existed modules refactoring
+There is already existed PFC asymmetric test cases which use PTF to send traffic.
+Test suite location is ```ansible\roles\test\files\saitests\pfc_asym.py```
+
+It requires several updates:
+
+1. Packets sending speed can be increased be using multiprocessing instead of multithreading library.
+
+```PfcAsymBaseTest.send```
+
+Use ```multiprocessing.Process``` instead of ```threading.Thread```.
+
+2. Looks like there is a redundancy in generating configuration file for ARP responder.
+
+```PfcAsymBaseTest.setUpArpResponder```
+
+Fix the loop to store file only once
+
+3. Need to remove dst port drop counters verification.
+
+```PfcAsymOffOnTxTest.runTest```
+
+Remove the following step which is defined by inline comment:
```

Comment:
Please explain in more details why we can avoid checking Tx drops:
- Purpose is to make the test more generic, right?
- For what test cases it is needed?
- In such cases the main goal is to cause DUT to generate PFC frames but not Tx drops, so it is enough to check PTF Tx counters on data ingress port and checking data egress drops is secondary and not necessary. Is that what you wanted to achieve?

Reply:
Fixed, please check the updated description